updater/fetchers/nodejs/nodejs.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2016 clair authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodejs
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/coreos/clair/database"
+	"github.com/coreos/clair/updater"
+	cerrors "github.com/coreos/clair/utils/errors"
+	"github.com/coreos/clair/utils/types"
+	"github.com/coreos/pkg/capnslog"
+)
+
+const (
+	url                  = "https://api.nodesecurity.io/advisories"
+	cveURLPrefix         = "http://cve.mitre.org/cgi-bin/cvename.cgi?name="
+	updaterFlag          = "nodejsUpdater"
+	defaultNodejsVersion = "all"
+	//FIXME: Add a suffix when an advisory is fixed `after` a certain version.
+	defaultVersionSuffix = "-1"
+)
+
+var log = capnslog.NewPackageLogger("github.com/coreos/clair", "updater/fetchers/nodejs")
+
+type nodejsAdvisory struct {
+	ID                 int      `json:"id"`
+	CreatedAt          string   `json:"created_at"`
+	UpdatedAt          string   `json:"updated_at"`
+	PublishDate        string   `json:"publish_date"`
+	Title              string   `json:"title"`
+	Author             string   `json:"author"`
+	ModuleName         string   `json:"module_name"`
+	CVES               []string `json:"cves"`
+	VulnerableVersions string   `json:"vulnerable_versions"`
+	PatchedVersions    string   `json:"patched_versions"`
+	Slug               string   `json:"slug"`
+	Overview           string   `json:"overview"`
+	Recommandation     string   `json:"recommandation"`
+	References         string   `json:"references"`
+	LegacySlug         string   `json:"legacy_slug"`
+	AllowedScopes      []string `json:"allowed_scopes"`
+	CvesVector         string   `json:"cves_vector"`
+	CvssScore          float32  `json:"cvss_score"`
+}
+
+type nodejsAdvisories struct {
+	Total   int              `json:"total"`
```

Comment:
I think we should omit to parse/store any non-required fields. This comment also applies to `nodejsAdvisory`.